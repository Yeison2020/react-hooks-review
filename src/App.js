import React from "react";
import PropTypes from "prop-types";
// import React, { Fragment } from "react";
// imr import react
function App({ salud, subtitle }) {
  const obj = {
    name: "Enmnauel",
    age: 26,
  };
  return (
    <>
      <h1>{salud}</h1>
      <p>{subtitle}</p>
    </>
  );
}

// This Allow me to expecify the data tyep of my object
App.propTypes = {
  salud: PropTypes.string.isRequired,
};

App.defaultProps = {
  subtitle: "Soy un subtitulo",
};

export default App;

{
  /* Comments React way */
}
{
  /* <p>{JSON.stringify(obj, null, 3)}</p> */
}
