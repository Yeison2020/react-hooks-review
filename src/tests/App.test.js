import React from "react";
import { getByText, render } from "@testing-library/react";
import App from "../App";

describe("Pruebas App", () => {
  test("Mostrar el mensaje Hola soy Yeison", () => {
    const salud = "Hola soy Yeison";

    const { getByAltText } = render(<App salud={salud} />);

    expect(getByText(salud)).toBeInTheDocument();
  });
});
