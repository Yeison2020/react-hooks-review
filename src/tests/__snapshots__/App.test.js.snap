// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas App Debe mostrar <App>Correctamente</App> 1`] = `
<Fragment>
  <h1>
    Hola soy Yeison !!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
